src/shared/clients/defaultLambdaClient.ts

```diff
@@ -62,7 +62,11 @@ export class DefaultLambdaClient implements LambdaClient {
 
         try {
             const response = await client.getFunction({ FunctionName: name }).promise()
-            getLogger().debug('GetFunction returned response: %O', response)
+            // prune `Code` from logs so we don't reveal a signed link to customer resources.
+            getLogger().debug('GetFunction returned response (code section pruned): %O', {
+                ...response,
+                Code: 'Pruned',
+            })
```

Comment:
seems extreme that we can't show debug information in `debug()` logs ...

Reply:
It's extreme but it also doesn't provide much information (aside from maybe the `RepositoryType`?: https://docs.aws.amazon.com/lambda/latest/dg/API_FunctionCodeLocation.html ); at worst, it represents a security concern and at least provides little value because the link will be inactive after 10 minutes.

We can check internally to see whether or not this is necessary.